linux_new/Jenkinsfile

```
@@ -0,0 +1,959 @@
+                        if (!fileExists(templatebase)) {
+                          error("Template File Not Found At : ${templatebase}")
+                        }
+
+                        def speccmd = "python3 linux_new/generate_spec.py \"${templatebase}\" \"${packagever}\" \"${packagearchDeb}\" \"${PKURL}\" \"${PCSUM}\" \"${PFILE}\" \"${outputfile}\" \"${currentDate}\" \"${PackageReleaseVersion}\" \"${upstreamversion}\" \"${packagever}\" \"${upstreamversion}\""
+                        // echo "Spec Command : ${speccmd}"
+                        def genresult = sh(script: speccmd, returnStatus: true)
+                        // echo "Result : ${genresult}"                        
+                        }
+
+                      } 
+                        
+                        if (DistArrayElement == "rhel" || DistArrayElement == "suse") {
+                        
+                        def verparts = Version.tokenize('.')
+                        if (verparts.size() == 2) {
+                            Version = "${verparts[0]}.${verparts[1]}.${TemurinVersion}"
+                            echo "Formatted Version: ${Version}"
+                        }                         
+
+                        if (Release == "8" ) {
+                            upstreamversion = params.TAG.replaceFirst("^jdk", "")
+                            // Extract components from original format
+                            def major = PVERS[0]           // Extract major version '8'
+                            def versparser = (PVERS =~ /(\d+)u(\d+)b(\d+)/)
+                            def minor = versparser[0][2]  // Extract '432'
+                            def build = versparser[0][3].replaceAll(/^0+/, '')    // Extract '6'
+                            def formattedBuild = build.toString()padLeft(2, '0')
+                            // Construct the new format
+                            packagever = "${major}.0.${minor}.0.${TemurinVersion}.${build}"
+                            changelogversion = "${major}.0.${minor}-b${formattedBuild}"
+                        } else {
+                            upstreamversion = params.TAG.replaceFirst("^jdk-", "")
+
+                            if (Version == "null") {
+                                packagever = "${Release}.0.0.0.${TemurinVersion}.${Build}"
+                                changelogversion = "${Release}.0.${TemurinVersion}+${Build}"
+                            } else {
+                                packagever = "${Release}.${Version}.${TemurinVersion}.${Build}"
+                                changelogversion = "${Release}.${Version}.${TemurinVersion}+${Build}"
+                            }
+                        }
+
+                        if (PARCH == 'arm') {
+                        // If JDK8 ARM32, Then Need To Deduce Correct Upstream
+                        upstreamversionARM32 = upstreamversionARM32.replaceFirst("^jdk", "")
```

Maybe stick both of these lines together?
```suggestion
                        upstreamversionARM32 = upstreamversionARM32.replaceFirst("^jdk", "").replaceAll(/-(jre|jdk)$/, "")
```

---

Done.